File: Random/binarySearch.cpp
#include <bits/stdc++.h>

using namespace std;

int binarySearch(vector<int> nums, int target) {
    int left = 0, right = nums.size()-1;

    while (left <= right) {
        int mid = (right - left)/2 + left;
        if (nums.at(mid) == target) {
            return mid;
        } else if (nums.at(mid) < target) {
            left = mid + 1;
        } else {
            right = mid - 1;
        }
    }
    
    return -1;
}

File: binarySearch.py
def binarySearch(lst, target):
    left, right = 0, len(lst)-1

    while left <= right:
        mid = (right - left)//2 + left

        if lst[mid] == target:
            return mid

        elif lst[mid] < target:
            left = mid + 1

        else:
            right = mid - 1

    return -1
